pkg/k8sutil/k8sutil.go

```diff
@@ -183,3 +184,44 @@ func isKubeMetaKind(kind string) bool {
 func isRunModeLocal() bool {
 	return os.Getenv(ForceRunModeEnv) == string(LocalRunMode)
 }
+
+// SupportsOwnerReference func checks whether a given dependent supports owner references, based in the owner.
```

Comment:
Can you add a bit more info about how this function works, since it's a user-facing library function?